<launch>

  <!-- run nodes on turtlebot machine -->
  <arg name="robot" default="0" doc="sets address for machine tag"/>

  <!-- rotation speed -->
  <arg name="frac_vel" default="0.5" doc="fraction of max rotation velocity"/>

  <!-- turtlebot3 diff drive params -->
  <rosparam command="load" file="$(find nuturtle_description)/config/diff_params.yaml" />

  <!-- basic remote -->
  <include file = "$(find nuturtle_robot)/launch/basic_remote.launch" >
    <arg name="robot" value="$(arg robot)" />
  </include>


  <!-- odometer 1 -->
  <!-- load params in these are private to the node  -->
  <node machine="turtlebot" name="odometer" pkg="rigid2d" type="odometer" output="screen" >
    <param name="odom_frame_id" value="odom" />
    <param name="body_frame_id" value="base_link" />
    <param name="left_wheel_joint" value="left_wheel_axle" />
    <param name="right_wheel_joint" value="right_wheel_axle" />
  </node>


  <!-- Interface with turtlebot3 -->
  <node machine="turtlebot" name="turtle_interface" pkg="nuturtle_robot" type="turtle_interface" output="screen" >
    <param name="left_wheel_joint" value="left_wheel_axle" />
    <param name="right_wheel_joint" value="right_wheel_axle" />
  </node>


  <!-- rotation node  -->
  <node machine="turtlebot" name="real_waypoint" pkg="nuturtle_robot" type="real_waypoint" output="screen" >
    <rosparam command="load" file="$(find nuturtle_robot)/config/real_waypoints.yaml" />
    <param name="frac_vel" type="double" value="$(eval arg('frac_vel'))" />
  </node>


  <!-- rviz -->
  <group if="$(eval arg('robot') != 0)">
  <node name="rviz" pkg="rviz" type="rviz" required="true" args="-d $(find tsim)/config/diff_drive_odom.rviz"/>
  </group>


  <!-- load model into parameter server -->
  <group if="$(eval arg('robot') != -1)">
    <param name="robot_description" command="$(find xacro)/xacro '$(find nuturtle_description)/urdf/diff_drive.urdf.xacro'"/>
  </group>


  <!-- robot state publisher -->
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"/>



</launch>
